// mineral-ui/FauxControl/styled.js
"use strict";

exports.__esModule = true;
exports.Control = exports.Underlay = exports.Suffix = exports.Prefix = exports.FauxControlRoot = void 0;

var _styledBase = _interopRequireDefault(require("@emotion/styled-base"));

var _isPropValid = _interopRequireDefault(require("@emotion/is-prop-valid"));

var _styles = require("../styles");

var _themes = require("./themes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var FauxControlRoot = (
/*#__PURE__*/
0, _styledBase.default)('div', {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'disabled' && (0, _isPropValid.default)(prop);
  },
  target: "e9lzmb40"
})(function (_ref) {
  var disabled = _ref.disabled,
      baseTheme = _ref.theme,
      variant = _ref.variant;
  var theme = (0, _themes.fauxControlTheme)(baseTheme);

  if (variant) {
    // prettier-ignore
    theme = _objectSpread({}, theme, {
      FauxControl_borderColor_hover: theme["borderColor_" + variant + "_hover"],
      FauxControl_boxShadow_active: "0 0 0 1px " + theme.boxShadow_focusInner + ", 0 0 0 2px " + theme["borderColor_" + variant + "_active"],
      FauxControl_boxShadow_focus: "0 0 0 1px " + theme.boxShadow_focusInner + ", 0 0 0 2px " + theme["borderColor_" + variant + "_focus"]
    });
  }

  return _objectSpread({}, (0, _styles.componentStyleReset)(baseTheme), {
    position: 'relative',
    zIndex: 1,
    '&:hover': {
      '& > div:last-child': {
        borderColor: !disabled ? theme.FauxControl_borderColor_hover : null
      }
    },
    '&:focus': {
      '& > div:last-child': {
        borderColor: !disabled ? theme.FauxControl_borderColor_focus : null,
        boxShadow: !disabled ? theme.FauxControl_boxShadow_focus : null
      }
    },
    '&:active': {
      '& > div:last-child': {
        borderColor: theme.FauxControl_borderColor_active,
        boxShadow: disabled ? 'none' : theme.FauxControl_boxShadow_active
      }
    }
  });
});
exports.FauxControlRoot = FauxControlRoot;
var Prefix = (
/*#__PURE__*/
0, _styledBase.default)('span', {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'size' && (0, _isPropValid.default)(prop);
  },
  target: "e9lzmb41"
})(function (_ref2) {
  var iconStart = _ref2.iconStart,
      size = _ref2.size,
      baseTheme = _ref2.theme;
  var theme = (0, _themes.fauxControlTheme)(baseTheme);
  var rtl = theme.direction === 'rtl';
  var fontSize = size === 'small' ? theme.FauxControl_fontSize_small : theme.FauxControl_fontSize;
  var marginWithIcon = (0, _styles.getNormalizedValue)(theme.FauxControl_paddingHorizontal, fontSize);
  var marginWithoutIcon = (0, _styles.getNormalizedValue)(parseFloat(theme.FauxControlIcon_marginHorizontal) / 2 + "em", fontSize);
  return _objectSpread({
    flex: '0 0 auto',
    fontSize: fontSize,
    marginLeft: rtl ? marginWithoutIcon : iconStart ? 0 : marginWithIcon,
    marginRight: rtl ? iconStart ? 0 : marginWithIcon : marginWithoutIcon,
    whiteSpace: 'nowrap'
  }, {
    display: "inline-block",
    maxWidth: "8em",
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap",
    wordWrap: "normal"
  });
});
exports.Prefix = Prefix;
var Suffix = (
/*#__PURE__*/
0, _styledBase.default)('span', {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'size' && (0, _isPropValid.default)(prop);
  },
  target: "e9lzmb42"
})(function (_ref3) {
  var iconEnd = _ref3.iconEnd,
      size = _ref3.size,
      baseTheme = _ref3.theme,
      variant = _ref3.variant;
  var theme = (0, _themes.fauxControlTheme)(baseTheme);
  var rtl = theme.direction === 'rtl';
  var fontSize = size === 'small' ? theme.FauxControl_fontSize_small : theme.FauxControl_fontSize;
  var marginWithIcon = (0, _styles.getNormalizedValue)(theme.FauxControl_paddingHorizontal, fontSize);
  var marginWithoutIcon = (0, _styles.getNormalizedValue)(parseFloat(theme.FauxControlIcon_marginHorizontal) / 2 + "em", fontSize);
  return _objectSpread({
    flex: '0 0 auto',
    fontSize: fontSize,
    marginLeft: rtl ? iconEnd || variant ? 0 : marginWithIcon : marginWithoutIcon,
    marginRight: rtl ? marginWithoutIcon : iconEnd || variant ? 0 : marginWithIcon,
    whiteSpace: 'nowrap'
  }, {
    display: "inline-block",
    maxWidth: "8em",
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap",
    wordWrap: "normal"
  });
});
exports.Suffix = Suffix;
var Underlay = (
/*#__PURE__*/
0, _styledBase.default)('div', {
  shouldForwardProp: function shouldForwardProp(prop) {
    return ['disabled', 'readOnly'].indexOf(prop) === -1 && (0, _isPropValid.default)(prop);
  },
  target: "e9lzmb43"
})(function (_ref4) {
  var disabled = _ref4.disabled,
      readOnly = _ref4.readOnly,
      baseTheme = _ref4.theme,
      variant = _ref4.variant;
  var theme = (0, _themes.fauxControlTheme)(baseTheme);
  return {
    backgroundColor: disabled || readOnly ? theme.input_backgroundColor_disabled : theme.FauxControl_backgroundColor,
    borderColor: variant && !disabled && !readOnly ? theme["borderColor_" + variant] : theme.FauxControl_borderColor,
    borderRadius: theme.FauxControl_borderRadius,
    borderStyle: 'solid',
    borderWidth: theme.FauxControl_borderWidth,
    bottom: 0,
    left: 0,
    position: 'absolute',
    right: 0,
    top: 0,
    zIndex: -1
  };
});
exports.Underlay = Underlay;
var Control = (
/*#__PURE__*/
0, _styledBase.default)('input', {
  shouldForwardProp: function shouldForwardProp(prop) {
    return ['as', 'prefix'].indexOf(prop) === -1;
  },
  target: "e9lzmb44"
})(function (_ref5) {
  var controlPropsIn = _ref5.controlPropsIn,
      controlSize = _ref5.controlSize,
      disabled = _ref5.disabled,
      hasPlaceholder = _ref5.hasPlaceholder,
      iconEnd = _ref5.iconEnd,
      iconStart = _ref5.iconStart,
      prefix = _ref5.prefix,
      readOnly = _ref5.readOnly,
      nonHtmlSize = _ref5.size,
      suffix = _ref5.suffix,
      baseTheme = _ref5.theme,
      variant = _ref5.variant;
  var theme = (0, _themes.fauxControlTheme)(baseTheme);
  var size = controlSize || nonHtmlSize;

  if (variant) {
    // prettier-ignore
    theme = _objectSpread({}, theme, {
      FauxControl_boxShadow_focus: "0 0 0 1px " + theme.boxShadow_focusInner + ", 0 0 0 2px " + theme["borderColor_" + variant + "_focus"]
    });
  }

  if (controlPropsIn.variant) {
    // prettier-ignore
    theme = _objectSpread({}, theme, {
      FauxControl_color: theme["color_" + controlPropsIn.variant]
    });
  }

  var rtl = theme.direction === 'rtl';
  var fontSize = size === 'small' ? theme.FauxControl_fontSize_small : theme.FauxControl_fontSize;
  var sizeAppropriateHorizontalPadding = size === 'small' || size === 'medium' ? theme.FauxControl_paddingHorizontal_small || theme.FauxControl_paddingHorizontal : theme.FauxControl_paddingHorizontal;
  var paddingWithoutIcon = (0, _styles.getNormalizedValue)(sizeAppropriateHorizontalPadding, fontSize);
  var color = theme.FauxControl_color;

  if (disabled) {
    color = theme.color_disabled;
  } else if (hasPlaceholder) {
    color = theme.FauxControl_color_placeholder;
  } else if (readOnly) {
    color = theme.FauxControl_color_readOnly;
  } // [1] - Safari and many Android browsers need this to apply the correct
  //       color to disabled controls


  var placeholderStyles = {
    color: theme.FauxControl_color_placeholder,
    WebkitTextFillColor: theme.FauxControl_color_placeholder,
    // [1]
    fontStyle: 'italic'
  };
  return {
    color: color,
    WebkitTextFillColor: color,
    // [1]
    fontSize: fontSize,
    fontStyle: hasPlaceholder ? 'italic' : null,
    outline: 0,
    paddingLeft: (iconStart || prefix) && !rtl || (iconEnd || variant || suffix) && rtl ? 0 : paddingWithoutIcon,
    paddingRight: (iconEnd || variant || suffix) && !rtl || (iconStart || prefix) && rtl ? 0 : paddingWithoutIcon,
    '&::placeholder': placeholderStyles,
    '&::-ms-input-placeholder': placeholderStyles,
    // Edge
    '&:-ms-input-placeholder': placeholderStyles,
    // IE 11
    '&::-ms-clear': {
      display: 'none'
    },
    '&:focus': {
      '& ~ div:last-child': {
        borderColor: theme.FauxControl_borderColor_focus,
        boxShadow: theme.FauxControl_boxShadow_focus
      }
    }
  };
});
exports.Control = Control;
