// mineral-ui/FauxControl/themes.js
"use strict";

exports.__esModule = true;
exports.fauxControlTheme = void 0;

var _styles = require("../styles");

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

// prettier-ignore
var fauxControlTheme = function fauxControlTheme(baseTheme) {
  return _objectSpread({
    FauxControl_backgroundColor: baseTheme.input_backgroundColor,
    FauxControl_borderColor: baseTheme.borderColor,
    FauxControl_borderColor_active: baseTheme.borderColor,
    FauxControl_borderColor_focus: baseTheme.borderColor,
    FauxControl_borderColor_hover: baseTheme.borderColor_theme_hover,
    FauxControl_borderRadius: baseTheme.borderRadius_1,
    FauxControl_borderWidth: '1px',
    FauxControl_boxShadow_active: "0 0 0 1px " + baseTheme.boxShadow_focusInner + ", 0 0 0 2px " + baseTheme.borderColor_theme_active,
    FauxControl_boxShadow_focus: "0 0 0 1px " + baseTheme.boxShadow_focusInner + ", 0 0 0 2px " + baseTheme.borderColor_theme_focus,
    FauxControl_color: baseTheme.color,
    FauxControl_color_placeholder: baseTheme.input_color_placeholder,
    FauxControl_color_readOnly: baseTheme.color_readOnly,
    FauxControl_fontSize: baseTheme.fontSize_ui,
    FauxControl_fontSize_small: (0, _styles.pxToEm)(12),
    FauxControl_paddingHorizontal: baseTheme.space_inset_md,
    FauxControl_paddingHorizontal_small: baseTheme.space_inset_sm,
    FauxControlIcon_marginHorizontal: baseTheme.space_inline_sm
  }, baseTheme);
};

exports.fauxControlTheme = fauxControlTheme;
