// mineral-ui/utils/index.js
"use strict";

exports.__esModule = true;
var _exportNames = {
  setFromArray: true,
  toArray: true,
  composeEventHandlers: true,
  isRenderProp: true,
  rtlTextAlign: true,
  withForwardRef: true,
  component: true
};
exports.component = exports.withForwardRef = exports.rtlTextAlign = exports.isRenderProp = exports.composeEventHandlers = exports.toArray = exports.setFromArray = void 0;

var _collections = require("./collections");

exports.setFromArray = _collections.setFromArray;
exports.toArray = _collections.toArray;

var _composeEventHandlers = _interopRequireDefault(require("./composeEventHandlers"));

exports.composeEventHandlers = _composeEventHandlers.default;

var _generateId = require("./generateId");

Object.keys(_generateId).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  exports[key] = _generateId[key];
});

var _isRenderProp = _interopRequireDefault(require("./isRenderProp"));

exports.isRenderProp = _isRenderProp.default;

var _propTypes = require("./propTypes");

exports.component = _propTypes.component;

var _rtlTextAlign = _interopRequireDefault(require("./rtlTextAlign"));

exports.rtlTextAlign = _rtlTextAlign.default;

var _withForwardRef = _interopRequireDefault(require("./withForwardRef"));

exports.withForwardRef = _withForwardRef.default;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
