// mineral-ui/Popover/PopoverContent.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _core = require("@emotion/core");

var _react = _interopRequireWildcard(require("react"));

var _withForwardRef = _interopRequireDefault(require("../utils/withForwardRef"));

var _PopoverArrow = _interopRequireDefault(require("./PopoverArrow"));

var _RtlPopper = _interopRequireDefault(require("./RtlPopper"));

var _styled = require("./styled");

var _constants = require("./constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var PopoverContent =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(PopoverContent, _Component);

  function PopoverContent() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = PopoverContent.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        children = _this$props.children,
        hasArrow = _this$props.hasArrow,
        modifiers = _this$props.modifiers,
        placement = _this$props.placement,
        positionFixed = _this$props.positionFixed,
        forwardedRef = _this$props.forwardedRef,
        subtitle = _this$props.subtitle,
        title = _this$props.title,
        restProps = _objectWithoutPropertiesLoose(_this$props, ["children", "hasArrow", "modifiers", "placement", "positionFixed", "forwardedRef", "subtitle", "title"]);

    var popperProps = {
      modifiers: modifiers,
      placement: placement,
      positionFixed: positionFixed
    };

    var _ref2 =
    /*#__PURE__*/
    (0, _core.jsx)(_styled.PopoverTitle, {
      subtitle: subtitle
    }, title);

    var _ref3 =
    /*#__PURE__*/
    (0, _core.jsx)(_styled.PopoverBlock, null, children);

    return (0, _core.jsx)(_RtlPopper.default, popperProps, function (_ref) {
      var arrowProps = _ref.arrowProps,
          outOfBoundaries = _ref.outOfBoundaries,
          placement = _ref.placement,
          popperRef = _ref.ref,
          style = _ref.style;

      var popoverContentWrapperProps = _objectSpread({
        'data-placement': placement,
        'data-out-of-boundaries': outOfBoundaries || undefined,
        ref: function ref(node) {
          forwardedRef && forwardedRef(node);
          popperRef(node);
        },
        style: style
      }, restProps);

      var popoverArrowProps = _objectSpread({
        size: _constants.ARROW_SIZE,
        placement: placement
      }, arrowProps);

      return (0, _core.jsx)(_styled.PopoverContentWrapper, popoverContentWrapperProps, title && _ref2, _ref3, hasArrow && (0, _core.jsx)(_PopoverArrow.default, popoverArrowProps));
    });
  };

  return PopoverContent;
}(_react.Component);

_defineProperty(PopoverContent, "displayName", 'PopoverContent');

var _default = (0, _withForwardRef.default)(PopoverContent);

exports.default = _default;
