// mineral-ui/Dropdown/themes.js
"use strict";

exports.__esModule = true;
exports.dropdownContentTheme = exports.dropdownTheme = void 0;

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var dropdownTheme = function dropdownTheme(baseTheme) {
  return _objectSpread({}, dropdownContentTheme(baseTheme), baseTheme);
};

exports.dropdownTheme = dropdownTheme;

var dropdownContentTheme = function dropdownContentTheme(baseTheme) {
  return _objectSpread({
    DropdownContent_backgroundColor: baseTheme.panel_backgroundColor,
    DropdownContent_borderColor: baseTheme.panel_borderColor,
    DropdownContent_borderRadius: baseTheme.borderRadius_1,
    DropdownContent_boxShadow: baseTheme.boxShadow_2,
    DropdownContent_margin: '5px',
    DropdownContent_zIndex: baseTheme.zIndex_100
  }, baseTheme);
};

exports.dropdownContentTheme = dropdownContentTheme;
