// mineral-ui/Dialog/Dialog.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _core = require("@emotion/core");

var _react = _interopRequireWildcard(require("react"));

var _exenv = require("exenv");

var _focusTrapReact = _interopRequireDefault(require("focus-trap-react"));

var _noScroll = _interopRequireDefault(require("no-scroll"));

var _Transition = _interopRequireDefault(require("react-transition-group/Transition"));

var _emotionTheming = require("emotion-theming");

var _utils = require("../utils");

var _children = require("../utils/children");

var _Button = _interopRequireDefault(require("../Button"));

var _Portal = _interopRequireDefault(require("../Portal"));

var _EventListener = _interopRequireDefault(require("../EventListener"));

var _constants = require("./constants");

var _styled = require("./styled");

var _themes = require("./themes");

var _DialogActions = _interopRequireDefault(require("./DialogActions"));

var _DialogBody = _interopRequireDefault(require("./DialogBody"));

var _DialogFooter = _interopRequireDefault(require("./DialogFooter"));

var _DialogHeader = _interopRequireDefault(require("./DialogHeader"));

var _DialogTitle = _interopRequireDefault(require("./DialogTitle"));

var _propTypes = require("./propTypes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Animation = (0, _emotionTheming.withTheme)(function (_ref) {
  var children = _ref.children,
      theme = _ref.theme,
      restProps = _objectWithoutPropertiesLoose(_ref, ["children", "theme"]);

  return (0, _core.jsx)(_Transition.default, _extends({
    appear: true,
    mountOnEnter: true,
    timeout: parseFloat((0, _themes.dialogTheme)(theme).Dialog_transitionDuration),
    unmountOnExit: true
  }, restProps), function (state) {
    return (0, _core.jsx)(_styled.DialogAnimate, {
      state: state
    }, children);
  });
});

var _ref3 =
/*#__PURE__*/
(0, _core.jsx)(_styled.DialogOverlay, null);

var Dialog =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(Dialog, _Component);

  function Dialog() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "state", {
      isExited: false,
      isExiting: false
    });

    _defineProperty(_assertThisInitialized(_this), "appNodes", void 0);

    _defineProperty(_assertThisInitialized(_this), "id", _this.props.id || "dialog-" + (0, _utils.generateId)());

    _defineProperty(_assertThisInitialized(_this), "dialogRoot", void 0);

    _defineProperty(_assertThisInitialized(_this), "dialogContent", void 0);

    _defineProperty(_assertThisInitialized(_this), "lastFocusedElement", void 0);

    _defineProperty(_assertThisInitialized(_this), "renderHeader", function () {
      var _this$props = _this.props,
          children = _this$props.children,
          closeButtonLabel = _this$props.closeButtonLabel,
          preventCloseButtonClose = _this$props.preventCloseButtonClose,
          showCloseButton = _this$props.showCloseButton,
          title = _this$props.title,
          variant = _this$props.variant;

      var closeButtonProps = _objectSpread({
        'aria-label': closeButtonLabel
      }, preventCloseButtonClose ? undefined : {
        onClick: _this.close
      }); // prettier-ignore


      var closeButton = showCloseButton ? (0, _core.jsx)(_styled.DialogCloseButton, closeButtonProps) : undefined;
      var header;
      var titleProps = {
        id: _this.getTitleId(),
        variant: variant
      };

      if (title) {
        header = (0, _core.jsx)(_DialogHeader.default, {
          closeButton: closeButton
        }, typeof title === 'string' ? (0, _core.jsx)(_DialogTitle.default, titleProps, title) : (0, _react.cloneElement)(title, titleProps));
      } else {
        var headerChild = (0, _children.findByType)(children, _DialogHeader.default);

        if (headerChild) {
          var headerChildProps = {
            titleProps: titleProps,
            closeButton: headerChild.props.closeButton || closeButton
          };
          header = (0, _react.cloneElement)(headerChild, headerChildProps);
        }
      }

      return header;
    });

    _defineProperty(_assertThisInitialized(_this), "renderBody", function () {
      var children = _this.props.children;
      var body;
      var bodyChild = (0, _children.findByType)(children, _DialogBody.default);

      if (bodyChild) {
        body = bodyChild;
      } else {
        var bodyChildren = (0, _children.excludeByType)(children, [_DialogHeader.default, _DialogFooter.default]);

        if (bodyChildren) {
          body = (0, _core.jsx)(_DialogBody.default, null, bodyChildren);
        }
      }

      return body;
    });

    _defineProperty(_assertThisInitialized(_this), "renderFooter", function () {
      var _this$props2 = _this.props,
          actions = _this$props2.actions,
          children = _this$props2.children,
          size = _this$props2.size,
          variant = _this$props2.variant;
      var footer;

      if (actions) {
        var keyedActions = actions.map(function (_ref2, index) {
          var text = _ref2.text,
              buttonSize = _ref2.size,
              restProps = _objectWithoutPropertiesLoose(_ref2, ["text", "size"]);

          var buttonProps = _objectSpread({
            size: buttonSize || (size === _constants.ACTIONS_SIZE.large ? _constants.ACTIONS_SIZE.large : _constants.ACTIONS_SIZE.medium)
          }, restProps);

          return (0, _core.jsx)(_Button.default, _extends({}, buttonProps, {
            key: index
          }), text);
        });
        footer = (0, _core.jsx)(_DialogFooter.default, null, (0, _core.jsx)(_DialogActions.default, {
          variant: variant
        }, keyedActions));
      } else {
        var footerChild = (0, _children.findByType)(children, _DialogFooter.default);

        if (footerChild) {
          footer = footerChild;
        }
      }

      return footer;
    });

    _defineProperty(_assertThisInitialized(_this), "getTitleId", function () {
      return _this.id + "-title";
    });

    _defineProperty(_assertThisInitialized(_this), "getContentId", function () {
      return _this.id + "-content";
    });

    _defineProperty(_assertThisInitialized(_this), "setContentRef", function (node) {
      _this.dialogContent = node;
    });

    _defineProperty(_assertThisInitialized(_this), "setRootRef", function (node) {
      _this.dialogRoot = node;
    });

    _defineProperty(_assertThisInitialized(_this), "setLastFocusedElement", function () {
      if (_exenv.canUseDOM) {
        var _global$document = global.document,
            activeElement = _global$document.activeElement,
            body = _global$document.body;
        _this.lastFocusedElement = activeElement && activeElement.focus ? activeElement : body;
      }
    });

    _defineProperty(_assertThisInitialized(_this), "restoreFocus", function () {
      _this.lastFocusedElement && _this.lastFocusedElement.focus();
    });

    _defineProperty(_assertThisInitialized(_this), "setInitialFocus", function () {
      _this.dialogRoot && _this.dialogRoot.focus();
    });

    _defineProperty(_assertThisInitialized(_this), "open", function () {
      var modeless = _this.props.modeless;

      _this.setLastFocusedElement();

      if (!modeless) {
        _this.setAppNode();

        _this.disableAppNode();

        if (_exenv.canUseDOM) {
          _noScroll.default.on();
        }
      }

      _this.setState({
        isExited: false
      });
    });

    _defineProperty(_assertThisInitialized(_this), "close", function () {
      _this.handleExiting();
    });

    _defineProperty(_assertThisInitialized(_this), "handleClick", function (event) {
      if (_this.isEventOutsideNode(event, _this.dialogContent)) {
        _this.close();
      }
    });

    _defineProperty(_assertThisInitialized(_this), "handleDocumentKeydown", function (event) {
      if (event.key.indexOf('Esc') === 0 && !event.defaultPrevented) {
        _this.close();
      }
    });

    _defineProperty(_assertThisInitialized(_this), "handleEntered", function () {
      !_this.props.disableFocusOnOpen && _this.setInitialFocus();
      _this.props.onOpen && _this.props.onOpen();
    });

    _defineProperty(_assertThisInitialized(_this), "handleExiting", function () {
      _this.setState({
        isExiting: true
      });
    });

    _defineProperty(_assertThisInitialized(_this), "handleExited", function () {
      var _this$props3 = _this.props,
          modeless = _this$props3.modeless,
          onClose = _this$props3.onClose;

      _this.setState({
        isExited: true,
        isExiting: false
      }, function () {
        if (!modeless) {
          if (_exenv.canUseDOM) {
            _noScroll.default.off();
          }

          _this.enableAppNode();
        }

        _this.restoreFocus();

        onClose && onClose();
      });
    });

    _defineProperty(_assertThisInitialized(_this), "isEventOutsideNode", function (event, node) {
      var target = event.target;
      return node && target instanceof Node && !node.contains(target);
    });

    _defineProperty(_assertThisInitialized(_this), "setAppNode", function () {
      var appSelector = _this.props.appSelector;

      if (appSelector && _exenv.canUseDOM) {
        _this.appNodes = Array.from(document.querySelectorAll(appSelector));

        if (!_this.appNodes.length) {
          throw new Error("[mineral-ui/Dialog]: Unable to find app node(s) using the appSelector of '" + appSelector + "'.");
        }
      }
    });

    _defineProperty(_assertThisInitialized(_this), "disableAppNode", function () {
      _this.appNodes && _this.appNodes.forEach(function (node) {
        return node.setAttribute('aria-hidden', 'true');
      });
    });

    _defineProperty(_assertThisInitialized(_this), "enableAppNode", function () {
      _this.appNodes && _this.appNodes.forEach(function (node) {
        return node.removeAttribute('aria-hidden');
      });
    });

    return _this;
  }

  var _proto = Dialog.prototype;

  _proto.componentDidUpdate = function componentDidUpdate(prevProps) {
    if (!prevProps.isOpen && this.props.isOpen) {
      this.open();
    }
  };

  _proto.render = function render() {
    var _this$props4 = this.props,
        children = _this$props4.children,
        closeOnClickOutside = _this$props4.closeOnClickOutside,
        closeOnEscape = _this$props4.closeOnEscape,
        disableFocusTrap = _this$props4.disableFocusTrap,
        isOpen = _this$props4.isOpen,
        hideOverlay = _this$props4.hideOverlay,
        modeless = _this$props4.modeless,
        ignoreOnClose = _this$props4.onClose,
        ignoreOnOpen = _this$props4.onOpen,
        size = _this$props4.size,
        title = _this$props4.title,
        usePortal = _this$props4.usePortal,
        restProps = _objectWithoutPropertiesLoose(_this$props4, ["children", "closeOnClickOutside", "closeOnEscape", "disableFocusTrap", "isOpen", "hideOverlay", "modeless", "onClose", "onOpen", "size", "title", "usePortal"]);

    var _this$state = this.state,
        isExited = _this$state.isExited,
        isExiting = _this$state.isExiting;

    if (isExited) {
      return null;
    }

    var headerId = this.getTitleId();
    var contentId = this.getContentId();
    var hasBody = (0, _children.findByType)(children, _DialogBody.default);
    var hasImmediatleTitle = (0, _children.findByType)(children, _DialogTitle.default);
    var hasHeader = title || (0, _children.findByType)(children, _DialogHeader.default);

    if (hasBody && hasImmediatleTitle) {
      throw new Error('[mineral-ui/Dialog] You must wrap DialogTitle in DialogHeader');
    }

    if (!hasHeader && !this.props['aria-label']) {
      throw new Error('[mineral-ui/Dialog] You must provide an accessible title via the title prop, the DialogTitle component, or the aria-label prop');
    }

    var rootProps = _objectSpread({
      'aria-labelledby': this.props['aria-label'] ? undefined : hasHeader ? headerId : contentId,
      'aria-modal': !modeless,
      id: this.id,
      ref: this.setRootRef,
      modeless: modeless,
      role: 'dialog',
      tabIndex: '-1'
    }, closeOnClickOutside && !modeless ? {
      onClick: this.handleClick
    } : undefined, restProps);

    var contentProps = {
      id: contentId,
      ref: this.setContentRef,
      role: 'document',
      size: size
    };
    var animationProps = {
      in: isOpen && !isExiting,
      onExiting: this.handleExiting,
      onExited: this.handleExited,
      onEntered: this.handleEntered
    };
    var focusTrapProps = {
      active: !disableFocusTrap && !modeless
    };
    var output = (0, _core.jsx)(Animation, animationProps, (0, _core.jsx)(_focusTrapReact.default, focusTrapProps, (0, _core.jsx)(_styled.DialogRoot, rootProps, !hideOverlay && !modeless && _ref3, (0, _core.jsx)(_styled.DialogIEWrapper, null, (0, _core.jsx)(_styled.DialogContent, contentProps, this.renderHeader(), this.renderBody(), this.renderFooter())), closeOnEscape && (0, _core.jsx)(_EventListener.default, {
      listeners: [{
        target: 'document',
        event: 'keydown',
        handler: this.handleDocumentKeydown
      }]
    }))));
    return usePortal ? (0, _core.jsx)(_Portal.default, null, output) : output;
  };

  return Dialog;
}(_react.Component);

exports.default = Dialog;

_defineProperty(Dialog, "displayName", 'Dialog');

_defineProperty(Dialog, "defaultProps", {
  closeButtonLabel: 'close',
  closeOnClickOutside: true,
  closeOnEscape: true,
  size: _constants.SIZE.medium,
  usePortal: true
});

Dialog.propTypes = process.env.NODE_ENV !== "production" ? _propTypes.dialogPropTypes : {};
