// mineral-ui/Dialog/styled.js
"use strict";

exports.__esModule = true;
exports.DialogTitleTitle = exports.DialogTitleRoot = exports.DialogRowRoot = exports.DialogOverlay = exports.DialogIEWrapper = exports.DialogHeaderRoot = exports.DialogFooterRoot = exports.DialogContent = exports.DialogCloseButton = exports.DialogBodyOverflowContainerWithShadows = exports.DialogBodyRoot = exports.DialogAnimate = exports.DialogActionsRoot = exports.DialogRoot = void 0;

var _core = require("@emotion/core");

var _styledBase = _interopRequireDefault(require("@emotion/styled-base"));

var _isPropValid = _interopRequireDefault(require("@emotion/is-prop-valid"));

var _react = _interopRequireDefault(require("react"));

var _withProps = _interopRequireDefault(require("recompose/withProps"));

var _styles = require("../styles");

var _themes = require("../themes");

var _emotion = require("../utils/emotion");

var _Button = _interopRequireDefault(require("../Button"));

var _Text = _interopRequireDefault(require("../Text"));

var _IconClose = _interopRequireDefault(require("../Icon/IconClose"));

var _OverflowContainerWithShadows2 = _interopRequireDefault(require("../OverflowContainer/OverflowContainerWithShadows"));

var _themes2 = require("./themes");

var _DialogRow = _interopRequireDefault(require("./DialogRow"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var DialogRoot = (
/*#__PURE__*/
0, _styledBase.default)('div', {
  target: "eom7rq70"
})(function (_ref) {
  var modeless = _ref.modeless,
      theme = _ref.theme;
  return _objectSpread({}, (0, _styles.componentStyleReset)(theme), {
    alignItems: 'center',
    bottom: 0,
    display: 'flex',
    justifyContent: 'center',
    left: 0,
    position: 'fixed',
    pointerEvents: modeless ? 'none' : undefined,
    right: 0,
    top: 0
  });
});
exports.DialogRoot = DialogRoot;
var DialogActionsRoot = (
/*#__PURE__*/
0, _styledBase.default)('div', {
  target: "eom7rq71"
})(function (_ref2) {
  var _notLastChil;

  var baseTheme = _ref2.theme;
  var theme = (0, _themes2.dialogActionsTheme)(baseTheme);
  var marginProperty = theme.direction === 'rtl' ? 'marginLeft' : 'marginRight';
  return {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'flex-end',
    '& > *:not(:last-child)': (_notLastChil = {}, _notLastChil[marginProperty] = theme.DialogActionsItem_margin, _notLastChil)
  };
});
exports.DialogActionsRoot = DialogActionsRoot;
var DialogAnimate = (
/*#__PURE__*/
0, _styledBase.default)('div', {
  target: "eom7rq72"
})(function (_ref3) {
  var state = _ref3.state,
      baseTheme = _ref3.theme;
  var theme = (0, _themes2.dialogTheme)(baseTheme);
  return {
    opacity: state === 'entered' ? 1 : 0,
    position: 'relative',
    transition: "opacity " + theme.Dialog_transitionDuration + " ease",
    willChange: 'opacity',
    zIndex: theme.Dialog_zIndex
  };
});
exports.DialogAnimate = DialogAnimate;
var DialogBodyRoot = (
/*#__PURE__*/
0, _styledBase.default)(_DialogRow.default, {
  target: "eom7rq73"
})(function (_ref4) {
  var _ref5;

  var baseTheme = _ref4.theme;
  var theme = (0, _themes2.dialogRowTheme)(baseTheme);
  var fontSize = theme.DialogRow_fontSize;
  var marginVertical = "" + (0, _styles.getNormalizedValue)(theme.DialogRow_marginVertical, fontSize);
  return _ref5 = {
    display: 'flex',
    flex: '1 1 auto',
    fontSize: fontSize,
    margin: 0,
    minHeight: '0%',
    // See: https://css-tricks.com/flexbox-truncated-text/#comment-1611744
    padding: 0
  }, _ref5['&:first-child' + _emotion.ignoreSsrWarning] = {
    marginTop: marginVertical
  }, _ref5['&:last-child'] = {
    marginBottom: marginVertical
  }, _ref5;
});
exports.DialogBodyRoot = DialogBodyRoot;
var DialogBodyThemedOverflowContainerWithShadows = (0, _themes.themed)(_OverflowContainerWithShadows2.default)(function (_ref6) {
  var baseTheme = _ref6.theme;
  return (0, _themes.mapComponentThemes)({
    name: 'DialogBody',
    theme: (0, _themes2.dialogBodyTheme)(baseTheme)
  }, {
    name: 'OverflowContainerWithShadows',
    theme: {}
  }, baseTheme);
});
var DialogBodyOverflowContainerWithShadows = (0, _withProps.default)({
  scrollY: true
})((
/*#__PURE__*/
0, _styledBase.default)(DialogBodyThemedOverflowContainerWithShadows, {
  target: "eom7rq74"
})(function (_ref7) {
  var _ref8;

  var baseTheme = _ref7.theme;
  var theme = (0, _themes2.dialogRowTheme)(baseTheme);
  var fontSize = theme.DialogRow_fontSize;
  var paddingHorizontal = "" + (0, _styles.getNormalizedValue)(theme.DialogRow_paddingHorizontal, fontSize);
  return _ref8 = {
    display: 'flex',
    flex: '1 1 auto',
    width: '100%',
    // OverflowContainerWithShadows > Scroller
    '& > div': {
      paddingLeft: paddingHorizontal,
      paddingRight: paddingHorizontal
    }
  }, _ref8['& > div > :first-child' + _emotion.ignoreSsrWarning] = {
    marginTop: 0
  }, _ref8['& > div > :last-child'] = {
    marginBottom: 0
  }, _ref8;
}));
exports.DialogBodyOverflowContainerWithShadows = DialogBodyOverflowContainerWithShadows;
var DialogThemedButton = (0, _themes.themed)(_Button.default)(function (_ref9) {
  var theme = _ref9.theme;
  return {
    ButtonIcon_color: theme.color
  };
});
var DialogCloseButton = (0, _withProps.default)({
  iconStart: (0, _core.jsx)(_IconClose.default, null),
  minimal: true,
  size: 'small',
  type: 'button'
})((
/*#__PURE__*/
0, _styledBase.default)(DialogThemedButton, {
  target: "eom7rq75"
})(function (_ref10) {
  var _ref11;

  var baseTheme = _ref10.theme;
  var theme = (0, _themes2.dialogTheme)(baseTheme);
  var marginProperty = theme.direction === 'rtl' ? 'marginRight' : 'marginLeft';
  return _ref11 = {}, _ref11[marginProperty] = theme.DialogCloseButton_margin, _ref11;
}));
exports.DialogCloseButton = DialogCloseButton;
var DialogContent = (
/*#__PURE__*/
0, _styledBase.default)('div', {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'size' && (0, _isPropValid.default)(prop);
  },
  target: "eom7rq76"
})(function (_ref12) {
  var size = _ref12.size,
      baseTheme = _ref12.theme;
  var theme = (0, _themes2.dialogTheme)(baseTheme);

  var getSizeStyles = function getSizeStyles(size) {
    var _ref13;

    var maxWidth = theme["DialogContent_maxWidth_" + size];
    var maxHeight = theme["DialogContent_maxHeight_" + size];
    var width = theme["DialogContent_width_" + size];
    var offsetVertical = theme.DialogContent_offsetVertical;
    var maxHeightNumber = parseFloat(maxHeight);
    var offsetVerticalNumber = parseFloat(offsetVertical);
    var minHeight = maxHeightNumber + 2 * offsetVerticalNumber + "em";
    return _ref13 = {
      maxWidth: maxWidth,
      width: width
    }, _ref13["@media(min-height: " + minHeight + ")"] = {
      maxHeight: maxHeight
    }, _ref13;
  };

  return _objectSpread({
    backgroundColor: theme.DialogContent_backgroundColor,
    border: "1px solid " + theme.DialogContent_borderColor,
    borderRadius: theme.DialogContent_borderRadius,
    boxShadow: theme.DialogContent_boxShadow,
    display: 'flex',
    flexDirection: 'column',
    maxHeight: theme.DialogContent_maxHeight,
    minWidth: theme.DialogContent_minWidth,
    pointerEvents: 'all',
    position: 'relative'
  }, getSizeStyles(size));
});
exports.DialogContent = DialogContent;
var DialogFooterRoot = (0, _withProps.default)({
  as: 'footer'
})((
/*#__PURE__*/
0, _styledBase.default)(_DialogRow.default, {
  target: "eom7rq77"
})({
  name: "1tu59u4",
  styles: "flex:0 0 auto;"
}));
exports.DialogFooterRoot = DialogFooterRoot;
var DialogHeaderRoot = (0, _withProps.default)({
  as: 'header'
})((
/*#__PURE__*/
0, _styledBase.default)(_DialogRow.default, {
  target: "eom7rq78"
})({
  name: "z49cen",
  styles: "display:flex;flex:0 0 auto;justify-content:space-between;"
}));
exports.DialogHeaderRoot = DialogHeaderRoot;
var DialogIEWrapper = (
/*#__PURE__*/
0, _styledBase.default)('div', {
  target: "eom7rq79"
})({
  name: "k008qs",
  styles: "display:flex;"
});
exports.DialogIEWrapper = DialogIEWrapper;
var DialogOverlay = (
/*#__PURE__*/
0, _styledBase.default)('div', {
  target: "eom7rq710"
})(function (_ref14) {
  var baseTheme = _ref14.theme;
  var theme = (0, _themes2.dialogTheme)(baseTheme);
  return {
    backgroundColor: theme.DialogOverlay_backgroundColor,
    bottom: 0,
    left: 0,
    overflow: 'hidden',
    position: 'absolute',
    right: 0,
    top: 0
  };
});
exports.DialogOverlay = DialogOverlay;
var DialogRowRoot = (
/*#__PURE__*/
0, _styledBase.default)('div', {
  target: "eom7rq711"
})(function (_ref15) {
  var baseTheme = _ref15.theme;
  var theme = (0, _themes2.dialogRowTheme)(baseTheme);
  var fontSize = theme.DialogRow_fontSize;
  return _objectSpread({}, (0, _styles.componentStyleReset)(baseTheme), {
    fontSize: fontSize,
    margin: (0, _styles.getNormalizedValue)(theme.DialogRow_marginVertical, fontSize) + " 0",
    outline: 0,
    padding: "0 " + (0, _styles.getNormalizedValue)(theme.DialogRow_paddingHorizontal, fontSize)
  });
});
exports.DialogRowRoot = DialogRowRoot;
var DialogTitleRoot = (
/*#__PURE__*/
0, _styledBase.default)('div', {
  target: "eom7rq712"
})(function (_ref16) {
  var _roleImg;

  var baseTheme = _ref16.theme,
      variant = _ref16.variant;
  var theme = (0, _themes2.dialogTitleTheme)(baseTheme);
  var marginProperty = theme.direction === 'rtl' ? 'marginLeft' : 'marginRight';
  return {
    color: variant ? theme["color_" + variant] : theme.DialogTitle_color,
    display: 'flex',
    '& > [role="img"]': (_roleImg = {
      color: variant ? theme["icon_color_" + variant] : null,
      flex: '0 0 auto'
    }, _roleImg[marginProperty] = theme.DialogTitleIcon_margin, _roleImg)
  };
});
exports.DialogTitleRoot = DialogTitleRoot;
var DialogTitleTitle = (
/*#__PURE__*/
0, _styledBase.default)(_Text.default, {
  target: "eom7rq713"
})({
  name: "mi52gy",
  styles: "color:inherit;flex:1 1 auto;"
});
exports.DialogTitleTitle = DialogTitleTitle;
