// mineral-ui/TextInput/propTypes.js
"use strict";

exports.__esModule = true;
exports.textInputPropTypes = void 0;

var _propTypes = require("prop-types");

var _constants = require("./constants");

var textInputPropTypes = {
  className: _propTypes.string,
  defaultValue: _propTypes.string,
  disabled: _propTypes.bool,
  htmlSize: (0, _propTypes.oneOfType)([_propTypes.number, _propTypes.string]),
  iconStart: _propTypes.element,
  iconEnd: _propTypes.element,
  inputRef: _propTypes.func,
  rootProps: _propTypes.object,
  invalid: _propTypes.bool,
  onChange: _propTypes.func,
  prefix: (0, _propTypes.oneOfType)([_propTypes.string, _propTypes.element]),
  readOnly: _propTypes.bool,
  required: _propTypes.bool,
  size: (0, _propTypes.oneOf)(Object.keys(_constants.SIZE)),
  suffix: (0, _propTypes.oneOfType)([_propTypes.string, _propTypes.element]),
  type: (0, _propTypes.oneOf)(Object.keys(_constants.TYPE)),
  value: _propTypes.string,
  variant: (0, _propTypes.oneOf)(Object.keys(_constants.VARIANT))
};
exports.textInputPropTypes = textInputPropTypes;
