// mineral-ui/es/Popover/PopoverContent.js
import { jsx as ___EmotionJSX } from "@emotion/core";

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { Component } from 'react';
import withForwardRef from "../utils/withForwardRef";
import PopoverArrow from "./PopoverArrow";
import Popper from "./RtlPopper";
import { PopoverContentWrapper, PopoverBlock, PopoverTitle } from "./styled";
import { ARROW_SIZE } from "./constants";

var PopoverContent =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(PopoverContent, _Component);

  function PopoverContent() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = PopoverContent.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        children = _this$props.children,
        hasArrow = _this$props.hasArrow,
        modifiers = _this$props.modifiers,
        placement = _this$props.placement,
        positionFixed = _this$props.positionFixed,
        forwardedRef = _this$props.forwardedRef,
        subtitle = _this$props.subtitle,
        title = _this$props.title,
        restProps = _objectWithoutPropertiesLoose(_this$props, ["children", "hasArrow", "modifiers", "placement", "positionFixed", "forwardedRef", "subtitle", "title"]);

    var popperProps = {
      modifiers: modifiers,
      placement: placement,
      positionFixed: positionFixed
    };

    var _ref2 =
    /*#__PURE__*/
    ___EmotionJSX(PopoverTitle, {
      subtitle: subtitle
    }, title);

    var _ref3 =
    /*#__PURE__*/
    ___EmotionJSX(PopoverBlock, null, children);

    return ___EmotionJSX(Popper, popperProps, function (_ref) {
      var arrowProps = _ref.arrowProps,
          outOfBoundaries = _ref.outOfBoundaries,
          placement = _ref.placement,
          popperRef = _ref.ref,
          style = _ref.style;

      var popoverContentWrapperProps = _objectSpread({
        'data-placement': placement,
        'data-out-of-boundaries': outOfBoundaries || undefined,
        ref: function ref(node) {
          forwardedRef && forwardedRef(node);
          popperRef(node);
        },
        style: style
      }, restProps);

      var popoverArrowProps = _objectSpread({
        size: ARROW_SIZE,
        placement: placement
      }, arrowProps);

      return ___EmotionJSX(PopoverContentWrapper, popoverContentWrapperProps, title && _ref2, _ref3, hasArrow && ___EmotionJSX(PopoverArrow, popoverArrowProps));
    });
  };

  return PopoverContent;
}(Component);

_defineProperty(PopoverContent, "displayName", 'PopoverContent');

export default withForwardRef(PopoverContent);
