// mineral-ui/OverflowContainer/OverflowContainer.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _core = require("@emotion/core");

var _react = _interopRequireWildcard(require("react"));

var _lodash = _interopRequireDefault(require("lodash.debounce"));

var _EventListener = _interopRequireDefault(require("../EventListener"));

var _styled = require("./styled");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var OverflowContainer =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(OverflowContainer, _Component);

  function OverflowContainer() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "state", {
      scrollable: false
    });

    _defineProperty(_assertThisInitialized(_this), "container", void 0);

    _defineProperty(_assertThisInitialized(_this), "setContainerRef", function (node) {
      var containerRef = _this.props.containerRef;
      _this.container = node;
      containerRef && containerRef(node);
    });

    _defineProperty(_assertThisInitialized(_this), "updateScrollable", function () {
      var _this$props = _this.props,
          scrollX = _this$props.scrollX,
          scrollY = _this$props.scrollY;
      var node = _this.container;

      if (node) {
        var scrollable = Boolean(scrollX && node.scrollWidth > node.clientWidth || scrollY && node.scrollHeight > node.clientHeight);

        if (_this.state.scrollable !== scrollable) {
          _this.setState({
            scrollable: scrollable
          });
        }
      }
    });

    _defineProperty(_assertThisInitialized(_this), "handleWindowResize", (0, _lodash.default)(_this.updateScrollable, 500));

    return _this;
  }

  var _proto = OverflowContainer.prototype;

  _proto.componentDidMount = function componentDidMount() {
    this.updateScrollable();
  };

  _proto.componentDidUpdate = function componentDidUpdate() {
    this.updateScrollable();
  };

  _proto.render = function render() {
    var _this$props2 = this.props,
        children = _this$props2.children,
        tabIndex = _this$props2.tabIndex,
        restProps = _objectWithoutPropertiesLoose(_this$props2, ["children", "tabIndex"]);

    var rootProps = _objectSpread({
      ref: this.setContainerRef
    }, tabIndex !== undefined ? {
      tabIndex: tabIndex
    } : this.state.scrollable ? {
      tabIndex: 0
    } : undefined, restProps);

    return (0, _core.jsx)(_styled.OverflowContainerRoot, rootProps, children, (0, _core.jsx)(_EventListener.default, {
      listeners: [{
        target: 'window',
        event: 'resize',
        handler: this.handleWindowResize
      }]
    }));
  };

  return OverflowContainer;
}(_react.Component);

exports.default = OverflowContainer;

_defineProperty(OverflowContainer, "displayName", 'OverflowContainer');
