// mineral-ui/es/Pagination/Pagination.js
import { jsx as ___EmotionJSX } from "@emotion/core";

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { PureComponent } from 'react';
import { withTheme } from 'emotion-theming';
import { FlexItem } from "../Flex";
import Pages from "./Pages";
import PageJumper from "./PageJumper";
import PageSizer from "./PageSizer";
import { PaginationRoot as Root } from "./styled";
import { paginationTheme } from "./themes";
import { SIZE } from "./constants";
import { paginationPropTypes } from "./propTypes";
export var Pagination =
/*#__PURE__*/
function (_PureComponent) {
  _inheritsLoose(Pagination, _PureComponent);

  function Pagination() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _PureComponent.call.apply(_PureComponent, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "pageJumperInput", void 0);

    _defineProperty(_assertThisInitialized(_this), "setPageJumperRef", function (node) {
      _this.pageJumperInput = node;
    });

    _defineProperty(_assertThisInitialized(_this), "getTotalPages", function (pageSize) {
      return Math.ceil(_this.props.totalCount / pageSize);
    });

    _defineProperty(_assertThisInitialized(_this), "handleClick", function (index) {
      _this.handlePageChange(index);
    });

    _defineProperty(_assertThisInitialized(_this), "handleIncrement", function (incrementForward, callback) {
      var currentPage = _this.props.currentPage;
      var nextPage = incrementForward ? currentPage + 1 : currentPage - 1;

      _this.handlePageChange(nextPage);

      callback && callback(nextPage);
    });

    _defineProperty(_assertThisInitialized(_this), "handlePageChange", function (currentPage) {
      _this.props.onPageChange(currentPage);

      if (_this.props.showPageJumper && _this.pageJumperInput && parseInt(_this.pageJumperInput.value) !== currentPage) {
        _this.pageJumperInput.value = '';
      }
    });

    _defineProperty(_assertThisInitialized(_this), "handlePageSizeChange", function (pageSize) {
      var _this$props = _this.props,
          onPageSizeChange = _this$props.onPageSizeChange,
          currentPage = _this$props.currentPage;
      onPageSizeChange && onPageSizeChange(pageSize);

      var totalPages = _this.getTotalPages(pageSize);

      if (currentPage > totalPages) {
        _this.handlePageChange(totalPages);
      }
    });

    return _this;
  }

  var _proto = Pagination.prototype;

  _proto.render = function render() {
    var _this$props2 = this.props,
        currentPage = _this$props2.currentPage,
        propMessages = _this$props2.messages,
        ignoreOnPageChange = _this$props2.onPageChange,
        ignoreOnPageSizeChange = _this$props2.onPageSizeChange,
        pageSize = _this$props2.pageSize,
        pageSizes = _this$props2.pageSizes,
        showPageJumper = _this$props2.showPageJumper,
        showPageSizer = _this$props2.showPageSizer,
        size = _this$props2.size,
        baseTheme = _this$props2.theme,
        totalCount = _this$props2.totalCount,
        restProps = _objectWithoutPropertiesLoose(_this$props2, ["currentPage", "messages", "onPageChange", "onPageSizeChange", "pageSize", "pageSizes", "showPageJumper", "showPageSizer", "size", "theme", "totalCount"]);

    var theme = paginationTheme(baseTheme);

    var messages = _objectSpread({}, Pagination.defaultProps.messages, propMessages);

    var rootProps = _objectSpread({
      'aria-label': messages.label,
      gutterWidth: theme.Pagination_gutterWidth
    }, restProps);

    var totalPages = this.getTotalPages(pageSize);

    var pagesProps = _objectSpread({
      currentPage: currentPage,
      handleClick: this.handleClick,
      handleIncrement: this.handleIncrement,
      messages: messages.pages,
      size: size,
      totalPages: totalPages
    }, restProps);

    var showPageSizerOrJumper = showPageSizer || showPageJumper;
    var pageJumperProps;

    if (showPageJumper) {
      pageJumperProps = {
        currentPage: currentPage,
        inputRef: this.setPageJumperRef,
        key: 'Page Jumper',
        messages: messages.pageJumper,
        onPageChange: this.handlePageChange,
        size: size,
        totalPages: totalPages,
        width: theme.PaginationPageJumper_width
      };
    }

    var pageSizerProps;

    if (showPageSizer) {
      pageSizerProps = {
        currentPage: currentPage,
        key: 'Page Sizer',
        messages: _objectSpread({
          category: messages.category
        }, messages.pageSizer),
        onPageSizeChange: this.handlePageSizeChange,
        pageSize: pageSize,
        pageSizes: pageSizes,
        size: size,
        totalCount: totalCount,
        totalPages: totalPages
      };
    }

    return ___EmotionJSX(Root, rootProps, showPageSizerOrJumper && ___EmotionJSX(FlexItem, {
      flex: true,
      grow: 1
    }, showPageSizer && ___EmotionJSX(FlexItem, null, ___EmotionJSX(PageSizer, pageSizerProps)), showPageJumper && ___EmotionJSX(FlexItem, {
      marginStart: "auto"
    }, ___EmotionJSX(PageJumper, pageJumperProps))), ___EmotionJSX(FlexItem, null, ___EmotionJSX(Pages, pagesProps)));
  };

  return Pagination;
}(PureComponent);

_defineProperty(Pagination, "displayName", 'Pagination');

_defineProperty(Pagination, "defaultProps", {
  messages: {
    category: 'rows',
    label: 'Pagination',
    pages: {
      next: 'Next',
      pageLabel: function pageLabel(isCurrentPage, isLastPage, page) {
        return (isCurrentPage ? 'Current page, ' : '') + (isLastPage ? 'Last page, ' : '') + page;
      },
      previous: 'Previous'
    },
    pageJumper: {
      label: 'Jump to page',
      placeholder: 'Page #'
    },
    pageSizer: {
      status: function status(category, first, last, total) {
        return first + "\u2013" + last + " of " + total + " " + category;
      },
      itemText: function itemText(pageSize) {
        return pageSize + " per page";
      }
    }
  },
  pageSizes: [10, 20, 50],
  showPageNumbers: true,
  size: SIZE.medium,
  visibleRange: 5
});

Pagination.propTypes = process.env.NODE_ENV !== "production" ? paginationPropTypes : {};
export default withTheme(Pagination);
