// mineral-ui/es/Table/TableHeaderRow.js
import { jsx as ___EmotionJSX } from "@emotion/core";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { Component } from 'react';
import deepEqual from 'react-fast-compare';
import TableSelectableCell from "./TableSelectableCell";
import TableHeaderCell from "./TableHeaderCell";
import TableSortableHeaderCell from "./TableSortableHeaderCell";
import TableRow from "./TableRow";

var TableHeaderRow =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(TableHeaderRow, _Component);

  function TableHeaderRow() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "toggle", function () {
      var toggle = _this.props.toggle;
      toggle && toggle();
    });

    return _this;
  }

  var _proto = TableHeaderRow.prototype;

  _proto.shouldComponentUpdate = function shouldComponentUpdate(nextProps) {
    return !deepEqual(this.props, nextProps);
  };

  _proto.render = function render() {
    var _this$props = this.props,
        checked = _this$props.checked,
        columns = _this$props.columns,
        indeterminate = _this$props.indeterminate,
        isSortable = _this$props.isSortable,
        tableSortable = _this$props.sortable,
        messages = _this$props.messages,
        toggle = _this$props.toggle,
        restProps = _objectWithoutPropertiesLoose(_this$props, ["checked", "columns", "indeterminate", "isSortable", "sortable", "messages", "toggle"]);

    var selectable = Boolean(toggle);
    var cells = columns.map(function (_ref) {
      var render = _ref.headerCell,
          content = _ref.content,
          columnSortable = _ref.sortable,
          key = _ref.key,
          label = _ref.label,
          restColumn = _objectWithoutPropertiesLoose(_ref, ["headerCell", "content", "sortable", "key", "label"]);

      if (typeof content !== 'string' && !label) {
        throw new Error('Columns with non-string content must define a `label` property.');
      }

      var cellProps = _objectSpread({
        children: content,
        key: key,
        label: label || (typeof content === 'string' ? content : undefined),
        messages: messages,
        render: render,
        sortable: columnSortable === false ? undefined : isSortable || columnSortable ? tableSortable : undefined
      }, restColumn);

      return cellProps.sortable ? ___EmotionJSX(TableSortableHeaderCell, _extends({
        name: key
      }, cellProps)) : ___EmotionJSX(TableHeaderCell, cellProps);
    });

    if (selectable) {
      cells.unshift(___EmotionJSX(TableSelectableCell, {
        key: messages.selectedRows,
        "aria-label": messages.selectedRows,
        isHeader: true,
        label: checked || indeterminate ? messages.deselectAllRows : messages.selectAllRows,
        checked: checked,
        indeterminate: indeterminate,
        onChange: this.toggle
      }));
    }

    var tableRowProps = _objectSpread({
      children: cells
    }, restProps);

    return ___EmotionJSX(TableRow, tableRowProps);
  };

  return TableHeaderRow;
}(Component);

_defineProperty(TableHeaderRow, "displayName", 'TableHeaderRow');

export { TableHeaderRow as default };
