// mineral-ui/es/Table/TableBase.js
import { jsx as ___EmotionJSX } from "@emotion/core";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { Component } from 'react';
import { generateId } from "../utils";
import TableContext from "./TableContext";
import TableDataRow from "./TableDataRow";
import TableHeaderRow from "./TableHeaderRow";
import TableHeader from "./TableHeader";
import TableTitle from "./TableTitle";
import { TableRoot as Root, TableBody, TableOverflowContainer as OverflowContainer } from "./styled";
import { DENSITY, TITLE_ELEMENT } from "./constants";

var TableBase =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(TableBase, _Component);

  function TableBase() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "id", _this.props.id || "table-" + generateId());

    _defineProperty(_assertThisInitialized(_this), "titleId", _this.id + "-title");

    return _this;
  }

  var _proto = TableBase.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        columns = _this$props.columns,
        data = _this$props.data,
        density = _this$props.density,
        hideHeader = _this$props.hideHeader,
        hideTitle = _this$props.hideTitle,
        highContrast = _this$props.highContrast,
        isSortable = _this$props.isSortable,
        messages = _this$props.messages,
        rowKey = _this$props.rowKey,
        scrollable = _this$props.scrollable,
        selectable = _this$props.selectable,
        sortable = _this$props.sortable,
        striped = _this$props.striped,
        title = _this$props.title,
        titleAppearance = _this$props.titleAppearance,
        titleAs = _this$props.titleAs,
        rootProps = _objectWithoutPropertiesLoose(_this$props, ["columns", "data", "density", "hideHeader", "hideTitle", "highContrast", "isSortable", "messages", "rowKey", "scrollable", "selectable", "sortable", "striped", "title", "titleAppearance", "titleAs"]);

    var appearanceProps = {
      density: density,
      highContrast: highContrast,
      striped: striped
    };
    var tableHeaderRowProps = {
      checked: selectable && selectable.all,
      columns: columns,
      indeterminate: selectable && selectable.some,
      isSortable: isSortable,
      messages: messages,
      sortable: sortable,
      toggle: selectable && selectable.toggleAll
    };

    var table = ___EmotionJSX(TableContext.Provider, {
      value: appearanceProps
    }, ___EmotionJSX(Root, rootProps, ___EmotionJSX(TableTitle, {
      appearance: titleAppearance,
      as: titleAs,
      hide: hideTitle,
      id: this.titleId
    }, title), ___EmotionJSX(TableHeader, {
      hide: hideHeader
    }, ___EmotionJSX(TableHeaderRow, tableHeaderRowProps)), ___EmotionJSX(TableBody, null, data.map(function (rowData, index) {
      var rowProps = {
        checked: selectable && selectable.isSelected(rowData),
        columns: columns,
        data: rowData,
        messages: messages,
        rowIndex: index,
        toggle: selectable && selectable.toggle
      };
      return ___EmotionJSX(TableDataRow, _extends({
        key: rowData[rowKey] || index
      }, rowProps));
    }))));

    if (scrollable) {
      var containerProps = {
        'aria-labelledby': this.titleId,
        role: 'group',
        scrollX: true
      };
      table = ___EmotionJSX(OverflowContainer, containerProps, table);
    }

    return table;
  };

  return TableBase;
}(Component);

_defineProperty(TableBase, "displayName", 'TableBase');

_defineProperty(TableBase, "defaultProps", {
  density: DENSITY.compact,
  scrollable: true,
  titleAs: TITLE_ELEMENT.h4
});

export { TableBase as default };
