// mineral-ui/es/Button/Button.js
import { jsx as ___EmotionJSX } from "@emotion/core";

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { Component, cloneElement } from 'react';
import withForwardRef from "../utils/withForwardRef";
import { Button as Root, Content, Inner } from "./styled";
import { ICON_SIZE, SIZE } from "./constants";
import { buttonPropTypes } from "./propTypes";

var Button =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(Button, _Component);

  function Button() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = Button.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        children = _this$props.children,
        disabled = _this$props.disabled,
        forwardedRef = _this$props.forwardedRef,
        iconStart = _this$props.iconStart,
        iconEnd = _this$props.iconEnd,
        _this$props$size = _this$props.size,
        size = _this$props$size === void 0 ? Button.defaultProps.size : _this$props$size,
        variant = _this$props.variant,
        restProps = _objectWithoutPropertiesLoose(_this$props, ["children", "disabled", "forwardedRef", "iconStart", "iconEnd", "size", "variant"]); // [1] Necessary when rendering an `a` element, which doesn't use `disabled`


    var rootProps = _objectSpread({
      'aria-disabled': disabled,
      // [1]
      disabled: disabled,
      ref: forwardedRef,
      size: size,
      tabIndex: disabled ? -1 : undefined,
      // [1]
      text: children,
      variant: variant
    }, restProps);

    var startIcon = iconStart ? cloneElement(iconStart, {
      size: ICON_SIZE[size],
      key: 'iconStart'
    }) : null;
    var endIcon = iconEnd ? cloneElement(iconEnd, {
      size: ICON_SIZE[size],
      key: 'iconEnd'
    }) : null;
    return ___EmotionJSX(Root, rootProps, ___EmotionJSX(Inner, null, startIcon, children && ___EmotionJSX(Content, {
      size: size
    }, children), endIcon));
  };

  return Button;
}(Component);

_defineProperty(Button, "displayName", 'Button');

_defineProperty(Button, "defaultProps", {
  size: SIZE.large
});

Button.propTypes = process.env.NODE_ENV !== "production" ? buttonPropTypes : {};
export default withForwardRef(Button);
