// mineral-ui/TextArea/styled.js
"use strict";

exports.__esModule = true;
exports.Input = exports.TextAreaRoot = void 0;

var _styledBase = _interopRequireDefault(require("@emotion/styled-base"));

var _isPropValid = _interopRequireDefault(require("@emotion/is-prop-valid"));

var _styles = require("../styles");

var _themes = require("../themes");

var _FauxControl = _interopRequireDefault(require("../FauxControl"));

var _themes2 = require("./themes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ThemedFauxControl = (0, _themes.themed)(_FauxControl.default)(function (_ref) {
  var baseTheme = _ref.theme;
  return (0, _themes.mapComponentThemes)({
    name: 'TextArea',
    theme: (0, _themes2.textAreaTheme)(baseTheme)
  }, {
    name: 'FauxControl',
    theme: {}
  }, baseTheme);
});
var TextAreaRoot = (
/*#__PURE__*/
0, _styledBase.default)(ThemedFauxControl, {
  target: "ecgwh2i0"
})({
  name: "6sun7c",
  styles: "align-items:center;cursor:text;display:flex;width:100%;"
});
exports.TextAreaRoot = TextAreaRoot;
var Input = (
/*#__PURE__*/
0, _styledBase.default)('textarea', {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'size' && (0, _isPropValid.default)(prop);
  },
  target: "ecgwh2i1"
})(function (_ref2) {
  var resizeable = _ref2.resizeable,
      size = _ref2.size,
      baseTheme = _ref2.theme;
  var theme = (0, _themes2.textAreaTheme)(baseTheme);
  var fontSize = size === 'small' ? theme.TextArea_fontSize_small : theme.TextArea_fontSize;
  var paddingVerticalNormalized = (0, _styles.getNormalizedValue)(theme["TextArea_paddingVertical_" + size], fontSize);
  return {
    backgroundColor: 'transparent',
    border: 0,
    boxShadow: 'none',
    flex: '1 1 auto',
    fontFamily: 'inherit',
    lineHeight: theme.lineHeight_prose,
    margin: theme.TextArea_borderWidth,
    // minHeight value is an attempt to display a single line of text.
    // It is needed when a user manually resizes a textarea.
    minHeight: parseFloat(paddingVerticalNormalized) * 2 + parseFloat(theme.TextArea_fontSize) * parseFloat(theme.lineHeight) + parseFloat((0, _styles.pxToEm)(parseFloat(theme.TextArea_borderWidth) * 2)) + parseFloat((0, _styles.pxToEm)(2)) + "em",
    minWidth: 0,
    outline: 0,
    paddingBottom: paddingVerticalNormalized,
    paddingTop: paddingVerticalNormalized,
    resize: resizeable ? 'vertical' : 'none',
    width: '100%'
  };
});
exports.Input = Input;
