// mineral-ui/TextArea/themes.js
"use strict";

exports.__esModule = true;
exports.textAreaTheme = void 0;

var _styles = require("../styles");

var _themes = require("../themes");

var _FauxControl = require("../FauxControl");

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
var textAreaTheme = function textAreaTheme(baseTheme) {
  return (0, _themes.mapComponentThemes)({
    name: 'FauxControl',
    theme: (0, _FauxControl.fauxControlTheme)(baseTheme)
  }, {
    name: 'TextArea',
    theme: {
      // The following padding values make appearances equivalent to TextInputs of same size when rows=1.
      // This enables usage of a TextArea as a single line input that can accept multiple lines of text.
      TextArea_paddingVertical_jumbo: (0, _styles.pxToEm)(14.5),
      TextArea_paddingVertical_large: (0, _styles.pxToEm)(8.5),
      TextArea_paddingVertical_medium: (0, _styles.pxToEm)(4.5),
      TextArea_paddingVertical_small: (0, _styles.pxToEm)(2)
    }
  }, baseTheme);
};

exports.textAreaTheme = textAreaTheme;
