// mineral-ui/Table/styled.js
"use strict";

exports.__esModule = true;
exports.TableTitleRoot = exports.TableSortableHeaderCellIconHolder = exports.TableSortableHeaderCellContent = exports.TableSortableHeaderCellButton = exports.TableSortableHeaderCellRoot = exports.PaddedCheckbox = exports.TableRowRoot = exports.TableHeaderCellRoot = exports.TableHeaderRoot = exports.TableCellRoot = exports.TableBody = exports.TableRoot = exports.TableOverflowContainer = void 0;

var _styledBase = _interopRequireDefault(require("@emotion/styled-base"));

var _isPropValid = _interopRequireDefault(require("@emotion/is-prop-valid"));

var _withProps = _interopRequireDefault(require("recompose/withProps"));

var _styles = require("../styles");

var _themes = require("../themes");

var _utils = require("../utils");

var _emotion = require("../utils/emotion");

var _Checkbox = _interopRequireDefault(require("../Checkbox"));

var _OverflowContainer = _interopRequireDefault(require("../OverflowContainer/OverflowContainer"));

var _themes2 = require("./themes");

var _TableHeaderCell = _interopRequireDefault(require("./TableHeaderCell"));

var _constants = require("./constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var REGEX_IS_EM_VALUE = /\d+em$/;

var tableCellStyles = function tableCellStyles(_ref) {
  var _ref2, _ref3;

  var density = _ref.density,
      highContrast = _ref.highContrast,
      noPadding = _ref.noPadding,
      textAlign = _ref.textAlign,
      baseTheme = _ref.theme;
  var theme = (0, _themes2.tableCellTheme)(baseTheme);
  var fontSize = theme.TableCell_fontSize;
  var rtl = theme.direction === 'rtl';
  var borderProperty = rtl ? 'borderRight' : 'borderLeft';
  var borderVertical = highContrast ? theme.TableCell_borderVertical_highContrast : theme.TableCell_borderVertical;
  var paddingHorizontal = (0, _styles.getNormalizedValue)(theme.TableCell_paddingHorizontal, fontSize);
  var paddingVertical = (0, _styles.getNormalizedValue)(density === 'spacious' ? theme.TableCell_paddingVertical_spacious : theme.TableCell_paddingVertical, fontSize);
  return _ref3 = {
    fontSize: fontSize,
    fontWeight: 'inherit',
    padding: noPadding ? 0 : paddingVertical + " " + paddingHorizontal,
    textAlign: (0, _utils.rtlTextAlign)(textAlign || 'start', theme.direction),
    verticalAlign: theme.TableCell_verticalAlign
  }, _ref3['&:not(:first-child)' + _emotion.ignoreSsrWarning] = (_ref2 = {}, _ref2[borderProperty] = borderVertical, _ref2), _ref3;
};

var TableOverflowContainer = (0, _themes.themed)(_OverflowContainer.default)(function (_ref4) {
  var baseTheme = _ref4.theme;
  return (0, _themes.mapComponentThemes)({
    name: 'Table',
    theme: (0, _themes2.tableTheme)(baseTheme)
  }, {
    name: 'OverflowContainer',
    theme: {}
  }, baseTheme);
});
exports.TableOverflowContainer = TableOverflowContainer;
var TableRoot = (
/*#__PURE__*/
0, _styledBase.default)('table', {
  target: "e1tsrciw0"
})(function (_ref5) {
  var theme = _ref5.theme;
  return _objectSpread({}, (0, _styles.componentStyleReset)(theme), {
    borderCollapse: 'collapse',
    borderSpacing: 0,
    width: '100%'
  });
});
exports.TableRoot = TableRoot;
var TableBody = (
/*#__PURE__*/
0, _styledBase.default)('tbody', {
  target: "e1tsrciw1"
})();
exports.TableBody = TableBody;
var TableCellRoot = (
/*#__PURE__*/
0, _styledBase.default)('td', {
  target: "e1tsrciw2"
})(tableCellStyles);
exports.TableCellRoot = TableCellRoot;
var TableHeaderRoot = (
/*#__PURE__*/
0, _styledBase.default)('thead', {
  target: "e1tsrciw3"
})(function (_ref6) {
  var hide = _ref6.hide,
      highContrast = _ref6.highContrast,
      baseTheme = _ref6.theme;
  var theme = (0, _themes2.tableHeaderTheme)(baseTheme);
  return hide ? {
    border: "0",
    clip: "rect(0 0 0 0)",
    clipPath: "inset(50%)",
    height: "1px",
    margin: "-1px",
    overflow: "hidden",
    padding: "0",
    position: "absolute",
    whiteSpace: "nowrap",
    width: "1px"
  } : {
    borderBottom: highContrast ? theme.TableHeader_borderBottom_highContrast : theme.TableHeader_borderBottom,
    borderTop: highContrast ? theme.TableHeader_borderTop_highContrast : theme.TableHeader_borderTop
  };
});
exports.TableHeaderRoot = TableHeaderRoot;
var TableHeaderCellRoot = (
/*#__PURE__*/
0, _styledBase.default)('th', {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'width' && (0, _isPropValid.default)(prop);
  },
  target: "e1tsrciw4"
})(function (_ref7) {
  var baseTheme = _ref7.theme,
      props = _objectWithoutPropertiesLoose(_ref7, ["theme"]);

  var theme = (0, _themes.mapComponentThemes)({
    name: 'TableHeaderCell',
    theme: (0, _themes2.tableHeaderCellTheme)(baseTheme)
  }, {
    name: 'TableCell',
    theme: {}
  }, baseTheme, ['TableHeaderCell_borderVertical', 'TableHeaderCell_fontSize', 'TableHeaderCell_paddingHorizontal', 'TableHeaderCell_paddingVertical', 'TableHeaderCell_verticalAlign']);
  return tableCellStyles(_objectSpread({
    theme: theme
  }, props));
}, function (_ref8) {
  var _before, _ref9, _ref10;

  var highContrast = _ref8.highContrast,
      maxWidth = _ref8.maxWidth,
      minWidth = _ref8.minWidth,
      baseTheme = _ref8.theme,
      width = _ref8.width;
  var theme = (0, _themes2.tableHeaderCellTheme)(baseTheme);
  var fontSize = theme.TableHeaderCell_fontSize;
  var rtl = theme.direction === 'rtl';
  var borderProperty = rtl ? 'borderRight' : 'borderLeft';
  var borderVertical = highContrast ? theme.TableHeaderCell_borderVertical_highContrast : theme.TableHeaderCell_borderVertical;
  var positionProperty = rtl ? 'right' : 'left';

  var getWidth = function getWidth(value, fontSize) {
    return REGEX_IS_EM_VALUE.test(value) ? (0, _styles.getNormalizedValue)(value, fontSize) : value;
  };

  return _ref10 = {
    fontWeight: theme.TableHeaderCell_fontWeight,
    maxWidth: getWidth(maxWidth, fontSize),
    minWidth: getWidth(minWidth, fontSize),
    position: 'relative',
    width: getWidth(width, fontSize),
    zIndex: 1
  }, _ref10['&:not(:first-child)' + _emotion.ignoreSsrWarning] = (_ref9 = {}, _ref9[borderProperty] = 0, _ref9['&::before'] = (_before = {}, _before[borderProperty] = borderVertical, _before.bottom = 0, _before.content = '""', _before[positionProperty] = 0, _before.position = 'absolute', _before.top = 0, _before.width = 0, _before.zIndex = -1, _before), _ref9), _ref10;
});
exports.TableHeaderCellRoot = TableHeaderCellRoot;
var TableRowRoot = (
/*#__PURE__*/
0, _styledBase.default)('tr', {
  target: "e1tsrciw5"
})(function (_ref11) {
  var _objectSpread2, _ref12;

  var highContrast = _ref11.highContrast,
      isSelected = _ref11.isSelected,
      baseTheme = _ref11.theme,
      striped = _ref11.striped;
  var theme = (0, _themes2.tableRowTheme)(baseTheme);
  return _objectSpread({
    backgroundColor: function () {
      if (isSelected) {
        if (highContrast) {
          return theme.TableRow_backgroundColor_highContrast_selected;
        }

        return theme.TableRow_backgroundColor_selected;
      }
    }()
  }, highContrast ? {
    borderBottom: theme.TableRow_borderHorizontal_highContrast
  } : {
    '&:not(:last-child)': {
      borderBottom: theme.TableRow_borderHorizontal
    }
  }, (_objectSpread2 = {
    '*:not(thead) > &:hover': {
      backgroundColor: function () {
        if (isSelected) {
          if (highContrast) {
            return theme.TableRow_backgroundColor_highContrast_selectedHover;
          }

          return theme.TableRow_backgroundColor_selectedHover;
        }

        return theme.TableRow_backgroundColor_hover;
      }()
    }
  }, _objectSpread2['&:nth-child(even):not(:hover)' + _emotion.ignoreSsrWarning] = {
    backgroundColor: !isSelected && striped ? theme.TableRow_backgroundColor_striped : null
  }, _objectSpread2), isSelected ? (_ref12 = {}, _ref12['& > td:first-child, & > th:first-child' + _emotion.ignoreSsrWarning] = {
    position: 'relative',
    '&::before': {
      backgroundColor: theme.color_theme_60,
      bottom: 0,
      content: '""',
      left: theme.direction !== 'rtl' ? 0 : null,
      right: theme.direction === 'rtl' ? 0 : null,
      position: 'absolute',
      top: 0,
      width: '4px'
    }
  }, _ref12) : undefined);
});
exports.TableRowRoot = TableRowRoot;
var PaddedCheckbox = (0, _withProps.default)({
  hideLabel: true
})((
/*#__PURE__*/
0, _styledBase.default)(_Checkbox.default, {
  target: "e1tsrciw6"
})(function (_ref13) {
  var density = _ref13.density,
      isHeader = _ref13.isHeader,
      baseTheme = _ref13.theme;
  var theme = isHeader ? (0, _themes2.tableHeaderCellTheme)(baseTheme) : (0, _themes2.tableCellTheme)(baseTheme);
  var themePrefix = isHeader ? 'TableHeaderCell' : 'TableCell';
  var paddingVertical = density === 'spacious' ? theme[themePrefix + "_paddingVertical_spacious"] : theme[themePrefix + "_paddingVertical"];
  return {
    padding: paddingVertical + " " + theme[themePrefix + "_paddingHorizontal"]
  };
}));
exports.PaddedCheckbox = PaddedCheckbox;

var tableSortableHeaderCellFocusStyles = function tableSortableHeaderCellFocusStyles(theme) {
  return {
    color: theme.TableSortableHeaderCell_color_focus,
    outline: theme.TableSortableHeaderCell_border_focus,
    outlineOffset: "-" + theme.TableSortableHeaderCell_border_focus.split(' ')[0]
  };
};

var TableSortableHeaderCellRoot = (0, _withProps.default)({
  noPadding: true
})((
/*#__PURE__*/
0, _styledBase.default)(_TableHeaderCell.default, {
  target: "e1tsrciw7"
})(function (_ref14) {
  var baseTheme = _ref14.theme;
  var theme = (0, _themes2.tableSortableHeaderCellTheme)(baseTheme);
  return {
    overflow: 'hidden',
    // [1]
    '&:hover': {
      color: theme.TableSortableHeaderCell_color_hover
    },
    '&:focus-within': tableSortableHeaderCellFocusStyles(theme)
  };
}));
exports.TableSortableHeaderCellRoot = TableSortableHeaderCellRoot;
var TableSortableHeaderCellButton = (0, _withProps.default)({
  as: 'button'
})((
/*#__PURE__*/
0, _styledBase.default)(_TableHeaderCell.default, {
  target: "e1tsrciw8"
})(function (_ref15) {
  var baseTheme = _ref15.theme;
  var theme = (0, _themes2.tableSortableHeaderCellTheme)(baseTheme);
  return {
    background: 'none',
    border: 0,
    color: 'inherit',
    cursor: 'pointer',
    fontSize: 'inherit',
    fontWeight: 'inherit',
    position: 'relative',
    whiteSpace: 'nowrap',
    width: '100%',
    '&:focus': tableSortableHeaderCellFocusStyles(theme),
    '*:focus-within > &:focus': {
      outline: 0
    },
    '&::-moz-focus-inner': {
      border: 0
    },
    // [1]
    '&::after': {
      bottom: '100%',
      content: '""',
      height: '10em',
      left: 0,
      position: 'absolute',
      right: 0
    }
  };
}));
exports.TableSortableHeaderCellButton = TableSortableHeaderCellButton;
var TableSortableHeaderCellContent = (
/*#__PURE__*/
0, _styledBase.default)('span', {
  target: "e1tsrciw9"
})({
  name: "156zkco",
  styles: "position:relative;white-space:normal;"
});
exports.TableSortableHeaderCellContent = TableSortableHeaderCellContent;
var TableSortableHeaderCellIconHolder = (
/*#__PURE__*/
0, _styledBase.default)('span', {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'direction' && (0, _isPropValid.default)(prop);
  },
  target: "e1tsrciw10"
})(function (_ref16) {
  var _ref17;

  var isSorted = _ref16.isSorted,
      direction = _ref16.direction,
      baseTheme = _ref16.theme;
  var theme = (0, _themes2.tableSortableHeaderCellTheme)(baseTheme);
  var iconAdjustment = (0, _styles.pxToEm)(2);
  var marginProperty = theme.direction === 'rtl' ? 'marginRight' : 'marginLeft';
  return _ref17 = {
    color: theme.icon_color,
    display: 'inline-block',
    height: theme.TableSortableHeaderCellIcon_size
  }, _ref17[marginProperty] = parseFloat(theme.space_inline_xxs) + parseFloat(iconAdjustment) + "em", _ref17.opacity = isSorted ? null : 0, _ref17.position = 'relative', _ref17.top = direction === _constants.SORT.ascending ? 2 : 1, _ref17.width = theme.TableSortableHeaderCellIcon_size, _ref17['& > [role="img"]'] = {
    margin: "-" + iconAdjustment
  }, _ref17['*:hover > button > &, button:focus > &'] = {
    color: 'inherit'
  }, _ref17;
});
exports.TableSortableHeaderCellIconHolder = TableSortableHeaderCellIconHolder;
var TableTitleRoot = (
/*#__PURE__*/
0, _styledBase.default)('caption', {
  target: "e1tsrciw11"
})(function (_ref18) {
  var hide = _ref18.hide,
      baseTheme = _ref18.theme;
  var theme = (0, _themes2.tableTitleTheme)(baseTheme);
  return _objectSpread({
    marginBottom: theme.TableTitle_marginBottom
  }, hide ? {
    border: "0",
    clip: "rect(0 0 0 0)",
    clipPath: "inset(50%)",
    height: "1px",
    margin: "-1px",
    overflow: "hidden",
    padding: "0",
    position: "absolute",
    whiteSpace: "nowrap",
    width: "1px"
  } : undefined);
}); // [1] Extends the clickable area of the Button to fill the entire cell

exports.TableTitleRoot = TableTitleRoot;
