// mineral-ui/Select/propTypes.js
"use strict";

exports.__esModule = true;
exports.selectPropTypes = void 0;

var _propTypes = require("prop-types");

var _propTypes2 = require("../Menu/propTypes");

var _constants = require("./constants");

var selectPropTypes = {
  data: _propTypes2.menuItemsPropType.isRequired,
  defaultHighlightedIndex: _propTypes.number,
  defaultIsOpen: _propTypes.bool,
  defaultSelectedItem: _propTypes.object,
  disabled: _propTypes.bool,
  highlightedIndex: _propTypes.number,
  id: _propTypes.string,
  invalid: _propTypes.bool,
  isOpen: _propTypes.bool,
  item: (0, _propTypes.oneOfType)([_propTypes.node, _propTypes.func]),
  itemKey: _propTypes.string,
  menu: (0, _propTypes.oneOfType)([_propTypes.node, _propTypes.func]),
  modifiers: _propTypes.object,
  name: _propTypes.string,
  onChange: _propTypes.func,
  onClose: _propTypes.func,
  onOpen: _propTypes.func,
  onSelect: _propTypes.func,
  placeholder: _propTypes.string,
  placement: (0, _propTypes.oneOf)(Object.keys(_constants.PLACEMENT)),
  positionFixed: _propTypes.bool,
  readOnly: _propTypes.bool,
  required: _propTypes.bool,
  selectedItem: _propTypes.object,
  size: (0, _propTypes.oneOf)(Object.keys(_constants.SIZE)),
  trigger: (0, _propTypes.oneOfType)([_propTypes.node, _propTypes.func]),
  triggerRef: _propTypes.func,
  usePortal: _propTypes.bool,
  variant: (0, _propTypes.oneOf)(Object.keys(_constants.VARIANT))
};
exports.selectPropTypes = selectPropTypes;
