// mineral-ui/Select/styled.js
"use strict";

exports.__esModule = true;
exports.TriggerContent = exports.Trigger = exports.SelectTriggerRoot = exports.contentWidthModifier = exports.SelectRoot = void 0;

var _styledBase = _interopRequireDefault(require("@emotion/styled-base"));

var _isPropValid = _interopRequireDefault(require("@emotion/is-prop-valid"));

var _styles = require("../styles");

var _themes = require("../themes");

var _emotion = require("../utils/emotion");

var _FauxControl = _interopRequireDefault(require("../FauxControl"));

var _Dropdown = _interopRequireDefault(require("../Dropdown/Dropdown"));

var _themes2 = require("./themes");

var _constants = require("./constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var ThemedDropdown = (0, _themes.themed)(_Dropdown.default)(function (_ref) {
  var baseTheme = _ref.theme;
  return (0, _themes.mapComponentThemes)({
    name: 'Select',
    theme: (0, _themes2.selectTheme)(baseTheme)
  }, {
    name: 'Dropdown',
    theme: {}
  }, baseTheme);
});
var SelectRoot = (
/*#__PURE__*/
0, _styledBase.default)(ThemedDropdown, {
  target: "evcziml0"
})({
  name: "16w6gzw",
  styles: "width:100%;& > span{width:100%;}"
});
exports.SelectRoot = SelectRoot;
var contentWidthModifier = {
  enabled: true,
  fn: function fn(data) {
    data.styles.minWidth = (0, _styles.pxToEm)(224);
    data.styles.width = (0, _styles.pxToEm)(data.offsets.reference.width);
    return data;
  }
};
exports.contentWidthModifier = contentWidthModifier;
var ThemedFauxControl = (0, _themes.themed)(_FauxControl.default)(function (_ref2) {
  var baseTheme = _ref2.theme;
  return (0, _themes.mapComponentThemes)({
    name: 'Select',
    theme: (0, _themes2.selectTriggerTheme)(baseTheme)
  }, {
    name: 'FauxControl',
    theme: {}
  }, baseTheme);
});
var SelectTriggerRoot = (
/*#__PURE__*/
0, _styledBase.default)(ThemedFauxControl, {
  target: "evcziml1"
})(function (_ref3) {
  var _roleImg;

  var disabled = _ref3.disabled,
      readOnly = _ref3.readOnly,
      selectedItemVariant = _ref3.selectedItemVariant,
      baseTheme = _ref3.theme,
      variant = _ref3.variant;
  var theme = (0, _themes2.selectTriggerTheme)(baseTheme);
  var rtl = theme.direction === 'rtl';
  return {
    alignItems: 'center',
    display: 'flex',
    width: '100%',
    // all icons
    '& [role="img"]': (_roleImg = {
      display: 'block',
      color: theme.SelectIcon_color,
      flex: '0 0 auto'
    }, _roleImg['&:first-child' + _emotion.ignoreSsrWarning] = {
      color: disabled || readOnly ? theme.color_disabled : selectedItemVariant ? theme["color_" + selectedItemVariant] : theme.SelectIcon_color,
      margin: "0 " + theme.SelectIcon_marginHorizontal
    }, _roleImg),
    // the arrow icon
    '& :not([role="img"]) ~ [role="img"]': {
      color: disabled || readOnly ? theme.color_disabled : variant ? theme["icon_color_" + variant] : theme.SelectIcon_color
    },
    // the variant icon
    '& :not([role="img"]) + [role="img"]:not(:last-of-type)': {
      color: disabled || readOnly ? theme.color_disabled : variant ? theme["color_" + variant] : selectedItemVariant ? theme["color_" + selectedItemVariant] : theme.SelectIcon_color,
      marginLeft: rtl ? null : theme.SelectIcon_marginHorizontal,
      marginRight: rtl ? theme.SelectIcon_marginHorizontal : null
    }
  };
});
exports.SelectTriggerRoot = SelectTriggerRoot;
var Trigger = (
/*#__PURE__*/
0, _styledBase.default)('div', {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'size' && (0, _isPropValid.default)(prop);
  },
  target: "evcziml2"
})(function (_ref4) {
  var size = _ref4.size,
      baseTheme = _ref4.theme;
  var theme = (0, _themes2.selectTriggerTheme)(baseTheme);
  var fontSize = size === _constants.SIZE.small ? theme.Select_fontSize_small : theme.Select_fontSize;
  return {
    alignItems: 'center',
    display: 'flex',
    flex: '1 1 auto',
    height: (0, _styles.getNormalizedValue)(theme["Select_height_" + size], fontSize),
    minWidth: 0
  };
});
exports.Trigger = Trigger;
var TriggerContent = (
/*#__PURE__*/
0, _styledBase.default)('span', {
  target: "evcziml3"
})(_objectSpread({}, {
  display: "inline-block",
  maxWidth: null,
  overflow: "hidden",
  textOverflow: "ellipsis",
  whiteSpace: "nowrap",
  wordWrap: "normal"
}, {
  userSelect: 'none',
  width: '100%'
}));
exports.TriggerContent = TriggerContent;
