// mineral-ui/es/Dialog/DialogBody.js
import { jsx as ___EmotionJSX } from "@emotion/core";

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { Component } from 'react';
import { DialogBodyRoot as Root, DialogBodyOverflowContainerWithShadows } from "./styled";
import ModifiersContext from "./ModifiersContext";
import { dialogBodyPropTypes } from "./propTypes";

var DialogBody =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(DialogBody, _Component);

  function DialogBody() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = DialogBody.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        children = _this$props.children,
        restProps = _objectWithoutPropertiesLoose(_this$props, ["children"]);

    var rootProps = _objectSpread({}, restProps);

    var modifiers = {
      flip: {
        boundariesElement: 'window'
      },
      preventOverflow: {
        escapeWithReference: true
      }
    };
    return ___EmotionJSX(Root, rootProps, ___EmotionJSX(DialogBodyOverflowContainerWithShadows, null, ___EmotionJSX(ModifiersContext.Provider, {
      value: modifiers
    }, children)));
  };

  return DialogBody;
}(Component);

_defineProperty(DialogBody, "displayName", 'DialogBody');

export { DialogBody as default };
DialogBody.propTypes = process.env.NODE_ENV !== "production" ? dialogBodyPropTypes : {};
