// mineral-ui/Truncate/styled.js
"use strict";

exports.__esModule = true;
exports.Tooltip = exports.TruncateRoot = void 0;

var _styledBase = _interopRequireDefault(require("@emotion/styled-base"));

var _cssSelectors = require("../utils/cssSelectors");

var _Tooltip2 = _interopRequireDefault(require("../Tooltip"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var TruncateRoot = (
/*#__PURE__*/
0, _styledBase.default)('span', {
  target: "e1i5p1da0"
})(function (_ref) {
  var _focus;

  var showTooltip = _ref.showTooltip,
      baseTheme = _ref.theme;
  return _objectSpread({}, showTooltip ? {
    pointerEvents: 'all'
  } : undefined, {
    display: "inline-block",
    maxWidth: "100%",
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap",
    wordWrap: "normal"
  }, {
    '&:focus': (_focus = {
      outline: "1px solid " + baseTheme.borderColor_theme_focus,
      outlineOffset: -1
    }, _focus[_cssSelectors.ie10Plus] = {
      outline: 'none'
    }, _focus)
  });
});
exports.TruncateRoot = TruncateRoot;
var Tooltip = (
/*#__PURE__*/
0, _styledBase.default)(_Tooltip2.default, {
  target: "e1i5p1da1"
})({
  name: "xs9fqv",
  styles: "white-space:normal;"
});
exports.Tooltip = Tooltip;
