// mineral-ui/es/Text/textWithThemeOverrides.js
import { jsx as ___EmotionJSX } from "@emotion/core";

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import { ThemeProvider } from "../themes";

var textWithThemeOverrides = function textWithThemeOverrides(_ref) {
  var appearance = _ref.appearance,
      _ref$as = _ref.as,
      as = _ref$as === void 0 ? 'h4' : _ref$as,
      children = _ref.children,
      textComponent = _ref.textComponent,
      displayName = _ref.displayName,
      theme = _ref.theme;
  var Text = textComponent;
  var textProps = {
    appearance: appearance,
    as: as,
    noMargins: true
  };

  var content = ___EmotionJSX(Text, textProps, children);

  if (theme[displayName + "_color"] !== undefined || theme[displayName + "_fontSize"] !== undefined || theme[displayName + "_fontWeight"] !== undefined) {
    var appliedAppearance = appearance || as; // prettier-ignore

    var getOverride = function getOverride(variable) {
      var _ref2;

      return theme[displayName + "_" + variable] !== undefined ? (_ref2 = {}, _ref2["Text_" + variable + "_" + appliedAppearance] = theme[displayName + "_" + variable], _ref2) : undefined;
    };

    var textTheme = _objectSpread({}, getOverride('color'), getOverride('fontSize'), getOverride('fontWeight'));

    content = ___EmotionJSX(ThemeProvider, {
      theme: textTheme
    }, content);
  }

  return content;
};

export default textWithThemeOverrides;
