// mineral-ui/Avatar/styled.js
"use strict";

exports.__esModule = true;
exports.AvatarRoot = void 0;

var _styledBase = _interopRequireDefault(require("@emotion/styled-base"));

var _isPropValid = _interopRequireDefault(require("@emotion/is-prop-valid"));

var _styles = require("../styles");

var _constants = require("./constants");

var _themes = require("./themes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var AvatarRoot = (
/*#__PURE__*/
0, _styledBase.default)('span', {
  shouldForwardProp: function shouldForwardProp(prop) {
    return ['color', 'shape', 'size'].indexOf(prop) === -1 && (0, _isPropValid.default)(prop);
  },
  target: "e15mnca50"
})(function (_ref) {
  var background = _ref.background,
      propColor = _ref.color,
      icon = _ref.icon,
      noBackground = _ref.noBackground,
      shape = _ref.shape,
      propSize = _ref.size,
      baseTheme = _ref.theme;
  var theme = (0, _themes.avatarTheme)(baseTheme);
  var color = propColor || theme.color_themePrimary;
  var size = theme["Avatar_size_" + propSize];
  return _objectSpread({}, (0, _styles.componentStyleReset)(baseTheme), {
    alignItems: 'center',
    backgroundColor: noBackground ? null : background || theme.color_theme_60,
    color: color,
    borderRadius: shape === _constants.SHAPE.square ? null : shape === _constants.SHAPE.rounded ? theme.borderRadius_1 : '100%',
    display: 'inline-flex',
    fontWeight: theme.Avatar_fontWeight,
    height: size,
    lineHeight: size,
    justifyContent: 'center',
    verticalAlign: 'middle',
    width: size,
    '& > abbr,& > span': {
      fontSize: icon ? null : theme["Avatar_fontSize_" + propSize]
    },
    '& > abbr': {
      textDecoration: 'none'
    },
    '& > img': {
      borderRadius: '100%',
      display: 'block',
      flex: '0 0 auto',
      maxWidth: '100%'
    },
    '& > [role="img"]': {
      color: color
    }
  });
});
exports.AvatarRoot = AvatarRoot;
