// mineral-ui/Flex/propTypes.js
"use strict";

exports.__esModule = true;
exports.flexItemPropTypes = exports.flexPropTypes = void 0;

var _propTypes = require("prop-types");

var _propTypes2 = require("../Box/propTypes");

var _constants = require("./constants");

var stringOrArrayOfStringsPropType = function stringOrArrayOfStringsPropType(constant) {
  var strings = Object.keys(constant);
  return (0, _propTypes.oneOfType)([(0, _propTypes.oneOf)(strings), (0, _propTypes.arrayOf)((0, _propTypes.oneOf)(strings))]);
};

var flexPropTypes = {
  alignItems: stringOrArrayOfStringsPropType(_constants.ALIGN_ITEMS),
  breakpoints: (0, _propTypes.arrayOf)((0, _propTypes.oneOfType)([_propTypes.number, _propTypes.string])),
  children: _propTypes.node,
  direction: stringOrArrayOfStringsPropType(_constants.DIRECTION),
  gutterWidth: (0, _propTypes.oneOfType)([_propTypes.number, _propTypes.string]),
  minWidth: _propTypes2.spacingPropType,
  justifyContent: stringOrArrayOfStringsPropType(_constants.JUSTIFY_CONTENT),
  wrap: (0, _propTypes.oneOfType)([_propTypes.bool, (0, _propTypes.arrayOf)((0, _propTypes.oneOfType)([_propTypes.bool]))])
};
exports.flexPropTypes = flexPropTypes;
var flexItemPropTypes = {
  alignSelf: stringOrArrayOfStringsPropType(_constants.ALIGN_SELF),
  breakpoints: (0, _propTypes.arrayOf)((0, _propTypes.oneOfType)([_propTypes.number, _propTypes.string])),
  flex: _propTypes.bool,
  grow: (0, _propTypes.oneOfType)([_propTypes.number, (0, _propTypes.arrayOf)(_propTypes.number)]),
  shrink: (0, _propTypes.oneOfType)([_propTypes.number, (0, _propTypes.arrayOf)(_propTypes.number)])
};
exports.flexItemPropTypes = flexItemPropTypes;
