// mineral-ui/Pagination/styled.js
"use strict";

exports.__esModule = true;
exports.PageJumperRoot = exports.PageJumperNumberInput = exports.PagesEllipsisButton = exports.PagesRoot = exports.PaginationRoot = void 0;

var _styledBase = _interopRequireDefault(require("@emotion/styled-base"));

var _withProps = _interopRequireDefault(require("recompose/withProps"));

var _styles = require("../styles");

var _themes = require("../themes");

var _Flex = _interopRequireWildcard(require("../Flex"));

var _TextInput = _interopRequireDefault(require("../TextInput"));

var _Button = _interopRequireDefault(require("../Button"));

var _themes2 = require("./themes");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var PaginationRoot = (0, _withProps.default)({
  as: 'nav',
  justifyContent: 'end'
})((
/*#__PURE__*/
0, _styledBase.default)(_Flex.default, {
  target: "e82prp50"
})(function (_ref) {
  var baseTheme = _ref.theme;
  var theme = (0, _themes2.paginationTheme)(baseTheme);
  return _objectSpread({}, (0, _styles.componentStyleReset)(baseTheme), {
    flexWrap: 'wrap-reverse',
    marginBottom: "-" + theme.Pagination_gutterWidth,
    '& > *': {
      marginBottom: theme.Pagination_gutterWidth
    }
  });
}));
exports.PaginationRoot = PaginationRoot;
var PagesRoot = (
/*#__PURE__*/
0, _styledBase.default)(_Flex.FlexItem, {
  target: "e82prp51"
})(function (_ref2) {
  var _notLastChild;

  var baseTheme = _ref2.theme;
  var theme = (0, _themes2.paginationTheme)(baseTheme);
  var rtl = theme.direction === 'rtl';
  var middleMargin = rtl ? 'marginLeft' : 'marginRight';
  return {
    '& > button, & > span': {
      '&:not(:last-child)': (_notLastChild = {}, _notLastChild[middleMargin] = theme.Pagination_gutterWidth, _notLastChild)
    }
  };
});
exports.PagesRoot = PagesRoot;
var ThemedButton = (0, _themes.themed)(_Button.default)(function (_ref3) {
  var theme = _ref3.theme;
  return {
    color_disabled: theme.color_theme
  };
});
var PagesEllipsisButton = (0, _withProps.default)({
  'aria-disabled': true,
  as: 'span',
  minimal: true,
  size: 'medium'
})((
/*#__PURE__*/
0, _styledBase.default)(ThemedButton, {
  target: "e82prp52"
})({
  name: "1ixbp0l",
  styles: "pointer-events:none;"
}));
exports.PagesEllipsisButton = PagesEllipsisButton;
var PageJumperNumberInput = (
/*#__PURE__*/
0, _styledBase.default)(_TextInput.default, {
  target: "e82prp53"
})({
  name: "u2av8b",
  styles: "& > input{-moz-appearance:textfield;&::-webkit-inner-spin-button, &::-webkit-outer-spin-button{-webkit-appearance:none;margin:0;}}"
});
exports.PageJumperNumberInput = PageJumperNumberInput;
var PageJumperRoot = (
/*#__PURE__*/
0, _styledBase.default)(_Flex.FlexItem, {
  target: "e82prp54"
})(function (_ref4) {
  var width = _ref4.width;
  return {
    width: width
  };
});
exports.PageJumperRoot = PageJumperRoot;
